tsdb/index/index.go

```diff
@@ -270,37 +331,51 @@ func (w *Writer) ensureStage(s indexWriterStage) error {
 	if w.stage == s {
 		return nil
 	}
+	if w.stage < s-1 {
+		// A stage has been skipped.
+		if err := w.ensureStage(s - 1); err != nil {
+			return err
+		}
+	}
 	if w.stage > s {
 		return errors.Errorf("invalid stage %q, currently at %q", s, w.stage)
 	}
 
 	// Mark start of sections in table of contents.
 	switch s {
 	case idxStageSymbols:
-		w.toc.Symbols = w.pos
+		w.toc.Symbols = w.f.pos
 		if err := w.startSymbols(); err != nil {
 			return err
 		}
 	case idxStageSeries:
 		if err := w.finishSymbols(); err != nil {
 			return err
 		}
-		w.toc.Series = w.pos
-
-	case idxStageLabelIndex:
-		w.toc.LabelIndices = w.pos
+		w.toc.Series = w.f.pos
 
 	case idxStageDone:
-		w.toc.Postings = w.pos
+		w.toc.LabelIndices = w.f.pos
+		// LabelIndices generation depends on the posting offset
+		// table produced at this stage.
+		if err := w.writePostingsToTmpFiles(); err != nil {
```

Comment:
Nice, the ordering of sections is unfortunate indeed (: Since we have TOC file... is changing the order really a breaking change? ;p

Reply:
Oh yes, we depend on it multiple times on reader even now, detecting how much bytes we should scan (: